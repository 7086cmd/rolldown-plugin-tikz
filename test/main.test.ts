import { expect, describe, it } from 'vitest'
import { rolldown } from 'rolldown'
import tikzPlugin from '../src/main.js'

describe('tikz', () => {
    it('should be handled default as text', async () => {
        const builder = await rolldown({
            input: ['./test/index.js'],
            platform: 'node',
            plugins: [tikzPlugin()],
        })
        let generated = await builder.generate()
        expect(generated.output[0].code).toMatchSnapshot()
    })

    it('should be handled as base64', async () => {
        const builder = await rolldown({
            input: ['./test/index.js'],
            plugins: [tikzPlugin()],
            platform: 'node',
            moduleTypes: {
                svg: 'base64'
            }
        })
        let generated = await builder.generate()
        expect(generated.output[0].code).toMatchSnapshot()
    })

    it('should be handled as dataurl', async () => {
        const builder = await rolldown({
            input: ['./test/index.js'],
            plugins: [tikzPlugin()],
            platform: 'node',
            moduleTypes: {
                svg: 'dataurl'
            }
        })
        let generated = await builder.generate()
        expect(generated.output[0].code).toMatchSnapshot()
    })
})
