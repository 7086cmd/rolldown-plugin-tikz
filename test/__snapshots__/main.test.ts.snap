// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tikz > should be handled as base64 1`] = `
"import assert from "node:assert";

//#region test/example.tikz
var example_default = "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"38.47\\" height=\\"38.47\\" viewBox=\\"-72 -72 28.853 28.853\\"><path fill=\\"none\\" stroke=\\"#000\\" stroke-miterlimit=\\"10\\" stroke-width=\\".4\\" d=\\"m-72.07-43.617 28.453-28.453\\"/></svg>";

//#endregion
//#region test/index.js
assert(typeof example_default === "string");

//#endregion"
`;

exports[`tikz > should be handled as dataurl 1`] = `
"import assert from "node:assert";

//#region test/example.tikz
var example_default = "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"38.47\\" height=\\"38.47\\" viewBox=\\"-72 -72 28.853 28.853\\"><path fill=\\"none\\" stroke=\\"#000\\" stroke-miterlimit=\\"10\\" stroke-width=\\".4\\" d=\\"m-72.07-43.617 28.453-28.453\\"/></svg>";

//#endregion
//#region test/index.js
assert(typeof example_default === "string");

//#endregion"
`;

exports[`tikz > should be handled default as text 1`] = `
"import assert from "node:assert";

//#region test/example.tikz
var example_default = "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"38.47\\" height=\\"38.47\\" viewBox=\\"-72 -72 28.853 28.853\\"><path fill=\\"none\\" stroke=\\"#000\\" stroke-miterlimit=\\"10\\" stroke-width=\\".4\\" d=\\"m-72.07-43.617 28.453-28.453\\"/></svg>";

//#endregion
//#region test/index.js
assert(typeof example_default === "string");

//#endregion"
`;
